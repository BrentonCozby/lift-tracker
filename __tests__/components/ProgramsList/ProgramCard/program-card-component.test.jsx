import React from 'react'
import 'raf/polyfill'
import renderer from 'react-test-renderer'
import Enzyme, { shallow, mount } from 'enzyme'
import EnzymeAdapter from 'enzyme-adapter-react-16'
import { Provider } from 'react-redux'
import { BrowserRouter as Router } from 'react-router-dom'
import configStore from '../../../../src/js/store.js'
import ConnectedProgramCard, { ProgramCard } from '../../../../src/js/components/ProgramsList/ProgramCard/program-card-component.jsx'

Enzyme.configure({ adapter: new EnzymeAdapter() })

describe('ProgramCard', () => {
    let initialState
    let initialProps
    let store

    function initializeComponent(props) {
        const $component = shallow(
            <Router>
                <ProgramCard {...props} />
            </Router>
        )

        const componentJSON = renderer.create(
            <Router>
                <ProgramCard {...props} />
            </Router>
        ).toJSON()

        return {
            props,
            $component,
            componentJSON
        }
    }

    function initializeConnectedComponent(props, state) {
        store = configStore(state)

        const $component = mount(
            <Provider store={store}>
                <Router>
                    <ConnectedProgramCard {...props} />
                </Router>
            </Provider>
        )

        const componentJSON = renderer.create(
            <Provider store={store}>
                <Router>
                    <ConnectedProgramCard {...props} />
                </Router>
            </Provider>
        ).toJSON()

        return {
            props,
            $component,
            componentJSON
        }
    }

    beforeEach(() => {
        initialProps = {
            programId: 'programId1234',
            title: 'program title',
            userId: 'userId1234',
            uid: 'uid1234'
        }
    })

    test('renders program card when user is logged in', () => {
        const { componentJSON } = initializeComponent(initialProps)

        expect(componentJSON).toMatchSnapshot()
    })

    test('renders program card when user is logged in (connected)', () => {
        initialState = {
            user: {
                uid: 'uid1234'
            }
        }

        delete initialProps.uid

        const { componentJSON } = initializeConnectedComponent(initialProps, initialState)

        expect(componentJSON).toMatchSnapshot()
    })
})