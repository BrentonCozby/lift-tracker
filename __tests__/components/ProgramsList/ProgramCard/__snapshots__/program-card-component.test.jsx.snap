// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgramCard renders program card when user is logged in (connected) 1`] = `
<a
  className="ProgramCard"
  href="programs/programId1234"
  onClick={[Function]}
>
  <h3
    className="ProgramCard-title"
  >
    program title
  </h3>
</a>
`;

exports[`ProgramCard renders program card when user is logged in 1`] = `
<a
  className="ProgramCard"
  href="programs/programId1234"
  onClick={[Function]}
>
  <h3
    className="ProgramCard-title"
  >
    program title
  </h3>
</a>
`;
